### DISCUSSION TITLE
Trying to match default site search results

### DISCUSSION THREAD

**Discussion by SJNBham** [microsoft-search/pnp-modern-search#2735] [Q&A] [answered]
Hi - we're new to editing result templates and are trying to get search results from a document library to display in list form similar to a standard site search. This includes the search term being highlighted and not just the summary of the document but the context of where the search term hits in the body of the document. We thought maybe adding [HitHighlightedSummary](https://microsoft-search.github.io/pnp-modern-search/v3/search-parts/templating/#handlebars-helpers) to the List results template might do this. Maybe our syntax is off or we're tweaking the wrong portion of the template:

Line 100: `<div>{{getSummary HitHighlightedSummary (slot item @root.slots.Summary)}}</div>`

The above isn't highlighting the searched term and appears to be using the summary value which seems different than what the site search results are returning. 

Thanks for any pointers you can send our way.

**Comment by dmoorjmalani** [comment]
Hi - We are seeing the same issue, the HitHighlightedSummary in out of the box search (first screenshot) is truncated and in PnP Modern Search webpart is coming as untruncated (second screenshot), we also tried querying SharePoint directly using SharePoint Rest API and in the results it returning untruncated version, 
`/_api/search/query?querytext=%27ContentTypeId:0x0101009D1CB255DA76424F860D91F20E6C41180066FB1E6E7FE76A49B84E1C328874F56B005D41A9E42481574AAA9CF191D729B7BB%20OR%20ContentTypeId:0x0101009D1CB255DA76424F860D91F20E6C411800CE99C9FAF0B2A44F900119F4FC93DE8C00725278535ED540458CCB07069E442E15%27`

When we query using keywords like Active Minds, it is returning truncated summary,
`/_api/search/query?querytext=%27active%20mind%27`

so, maybe it is related to [this](https://admin.microsoft.com/AdminPortal/Home?#/servicehealth/:/alerts/MO494352) Service Issue open in admin center, not really sure, any guidance is much appreciated. Thank you!!!


![image](https://user-images.githubusercontent.com/6703440/216387876-cc98fb32-6f90-4f61-a63c-a082fbb424c0.png)
![image](https://user-images.githubusercontent.com/6703440/216387953-b6ed397d-352b-439e-954b-b4a053bd578a.png)

**Comment by SJNBham** [comment]
Here's a comparison of what we're seeing (site search results and then search webpart results). We're trying to get the webpart search results to be closer to the site search results with the hit highlighting of the search term and pulling in what seems like a different summary that includes more sections of the document contents than the hit highlight summary value. 



<img width="806" alt="site-search" src="https://user-images.githubusercontent.com/11393597/216388010-f47c5ffc-66b1-4cb8-ad0a-a82e550b79a6.png">
<img width="918" alt="search-webpart" src="https://user-images.githubusercontent.com/11393597/216388018-d8e87322-4159-40d0-a0da-ff00caf3cc01.png">

**Comment by corriespondent** [comment]
@SJNBham - try this in the result template:
`<div>{{getSummary (slot item @root.slots.Summary)}}</div>`
Then in the layout slot, map the Summary slot name to "HitHighlightedSummary" in the Slot field:
![image](https://user-images.githubusercontent.com/23064475/216483595-e31c047b-9eba-42bf-b1a7-6968ef5414c9.png)

@dmoorjmalani and I are both seeing the same issue with what gets returned in HitHighlightedSummary, however, so I'm curious if you have the same issue once you update your layout slots and result template.

**Comment by SJNBham** [comment]
Hi @corriespondent - the settings you list are the default for using the List search result layout and out of the box Layout slots mapping. When we tried reverting back to these default settings, we don't get a difference in behavior. We still don't have hit highlighting and the summary isn't the same as the site search results are showing.

**Comment by SJNBham** [comment]
Just to clarify, we're using version 4.8.0 and we only see reference to the {{getSummary HitHighlightedSummary}} handlebar helper in the V3 documentation. Maybe this was deprecated with V4?

**Comment by corriespondent** [comment]
@SJNBham we are using 4.8.0 as well, and the default setting for List/layout slots shows the proper highlighting on a dev tenant when I run a keyword search (all I did was switch to List display and link the search box with the results web part, no other configuration):
![image](https://user-images.githubusercontent.com/23064475/216633961-7a9fdb7a-8044-48b9-8c2c-0e68bf7d2065.png)

When I run a star search on this tenant, the summaries display the same way for both OOTB search and PnP search:
![image](https://user-images.githubusercontent.com/23064475/216633865-8566a034-4710-495f-b8c5-4354de9c8d6b.png)

But to clarify -- my handlebar template (unchanged) shows the reference to the layout slot called "Summary," shown in the next screenshot. Directly referring to the HitHighlightedSummary property is in v3, which did not have layout slots. 
![image](https://user-images.githubusercontent.com/23064475/216634971-ca549804-0650-4fbc-85ab-711682e4100e.png)

I think the issue in your original post might have been that you had both HitHighlightedSummary and the layout slot in your code -- maybe?
`<div>{{getSummary HitHighlightedSummary (slot item @root.slots.Summary)}}</div>`

**Comment by SJNBham** [comment, marked as answer]
Hi @corriespondent - thanks for confirming the default settings show the hit highlighting. We tried adding just the default search box and results web part and eventually discovered our error. It appears the reason the hit highlighting isn't showing up for us is because we modified the Query template thinking we could add KQL query filters in that area. Reading over the V4 documentation more, it looks like we should have been using FQL and placing it in the Refinement Filters field on the results web part. We're new to FQL and its syntax so our initial testing of that field didn't work. Everything's working now and the results are matching what the site search using the KQL filter returns. So, to get a search result that shows a single document library with hit highlighting add something like the following to the Refinement filters field on the results web part:

Path:equals("https://my.sharepoint.com/sites/SiteName/DocumentLibraryName/*")

<img width="229" alt="image" src="https://user-images.githubusercontent.com/11393597/216666205-cb91ee28-0fe1-4e72-9ae3-7ea549bcd9bb.png">


Thanks @corriespondent and @dmoorjmalani for pitching in on helping us resolve this.

**Reply by corriespondent** [reply]
@SJNBham glad that you worked it out!

For the record - the comment from @dmoorjmalani is an issue we're seeing on a couple different [dev and production] tenants, but @wobba confirmed on Twitter that it is a SharePoint bug.
https://twitter.com/mikaelsvenson/status/1621419104744755208

**Reply by wobba** [reply]
@SJNBham curious as to why you're not just writing your path filter as KQL in the template settings instead of FQL in the filters part? :) Both works, but KQL is easier. `path:https://my.sharepoint.com/sites/SiteName/DocumentLibraryName/`

**Reply by wobba** [reply]
@dmoorjmalani do you still see long summaries without a keyword?

**Reply by SJNBham** [reply]
Hi @wobba - That is odd. We must have been missing something in the KQL syntax or experimenting with other settings as the hit highlighting wasn't working when we used the Query template field before. But trying it today it clearly seems to work. Thanks for the follow up.
